pkg/yttlibrary/ip.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package yttlibrary
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/k14s/starlark-go/starlark"
+	"github.com/k14s/starlark-go/starlarkstruct"
+	"github.com/k14s/ytt/pkg/orderedmap"
+	"github.com/k14s/ytt/pkg/template/core"
+)
+
+var (
+	IPAPI = starlark.StringDict{
+		"ip": &starlarkstruct.Module{
+			Name: "ip",
+			Members: starlark.StringDict{
+				"parse_addr": starlark.NewBuiltin("ip.parse_addr", core.ErrWrapper(ipModule{}.ParseAddr)),
+				"parse_cidr": starlark.NewBuiltin("ip.parse_cidr", core.ErrWrapper(ipModule{}.ParseCIDR)),
+			},
+		},
+	}
+)
+
+type ipModule struct{}
+
+// AddrValue stores a parsed IP
+type AddrValue struct {
+	addr                 net.IP
+	*core.StarlarkStruct // TODO: keep authorship of the interface by delegating instead of embedding
+}
+
+func (m ipModule) ParseAddr(thread *starlark.Thread, f *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	if args.Len() != 1 {
+		return starlark.None, fmt.Errorf("expected exactly one argument")
+	}
+
+	ipStr, err := core.NewStarlarkValue(args.Index(0)).AsString()
+	if err != nil {
+		return starlark.None, err
+	}
+
+	parsedIP := net.ParseIP(ipStr)
+	if parsedIP == nil {
+		return starlark.None, fmt.Errorf("invalid IP address: %s", ipStr)
+	}
+	return (&AddrValue{parsedIP, nil}).AsStarlarkValue(), nil
+}
+
+func (av *AddrValue) Type() string { return "@ytt:ip.addr" }
+
+func (av *AddrValue) AsStarlarkValue() starlark.Value {
+	m := orderedmap.NewMap()
+	m.Set("is_ipv4", starlark.NewBuiltin("addr.is_ipv4", core.ErrWrapper(av.IsIPv4)))
+	m.Set("is_ipv6", starlark.NewBuiltin("addr.is_ipv6", core.ErrWrapper(av.IsIPv6)))
```

Comment:
```suggestion
	m.Set("is_ipv4", starlark.NewBuiltin("ip.addr.is_ipv4", core.ErrWrapper(av.IsIPv4)))
	m.Set("is_ipv6", starlark.NewBuiltin("ip.addr.is_ipv6", core.ErrWrapper(av.IsIPv6)))
```

Let's namespace the methods to modules to ensure no inadvertent collisions.  (In retrospect, we should have done the same for the methods on `@ytt:url.url`).

Reply:
fixed